src/configPaths.ts

```diff
@@ -25,24 +25,30 @@ export type NonConfigEnv = {
 }
 
 export const configPaths = <T extends NonConfigEnv>(
-  process: T
+  process: T,
+  subPath?: string
 ): ConfigTypes => {
   const baseDIR = baseConfigPath(process)
+  const subDir = path.parse(subPath || '').dir
   const defaultConfig = path.resolve(
     process.cwd(),
+    subDir,
     `${baseDIR}/${DEFAULT_FILENAME}.json`
   )
   const envConfig = path.resolve(
     process.cwd(),
+    subDir,
     `${baseDIR}/env/${process.env['NODE_ENV'] || DEFAULT_FILENAME}.json`
   )
   const deploymentConfig = path.resolve(
     process.cwd(),
+    subDir,
     `${baseDIR}/deployment/${process.env['DEPLOYMENT'] ||
       DEFAULT_FILENAME}.json`
   )
   const userConfig = path.resolve(
     process.cwd(),
+    subDir,
```

Comment:
`path.resolve` should take in only two params — 

1. The path where the config dir was found.
2. The path generated based on the env variables.